### PR TITLE
Veda top album 

### LINKED ISSUE #67: Implementation of Top Album use case and interface adapters
implement the top album use case and interface adapters, following clean architecture and solid principles.

### PR DESCRIPTION
**Pull request by coderveda27**
I worked on the interface adapters and completed controller, presenter, state, view model for my use case. I also reimplemented my input boundary and input data for my use case. I may add more methods to the state later as needed and I need to implement the success view in the presenter still.  Also, I noticed I forgot to implement the output boundary interface in the interactor class which caused the error :)

Also tackled debugging the shared top songs use case. I made a more so temporary fix by just adding the third param and initializing it but I will write the logic to update the third param with a value rather than it just being null. Going to do that soon.

### CODE REVIEW

**Comment by lunazay** [comment]
all reviews look good, maybe there are some style suggestions that could make the format better :)

File: src/interface_adapter/top_album/TopAlbumViewModel.java
Comment (jackwiebe0) [inline src/interface_adapter/top_album/TopAlbumViewModel.java:1]: I have my view name as lowercase, but here it's uppercase. We could resolve this in the future.

File: src/interface_adapter/top_album/TopAlbumState.java
Comment (jackwiebe0) [inline src/interface_adapter/top_album/TopAlbumState.java:1]: Maybe add a more descriptive error name?

**Review by lunazay — APPROVED** [review]

File: src/interface_adapter/top_album/TopAlbumViewModel.java
Comment (coderveda27) [reply src/interface_adapter/top_album/TopAlbumViewModel.java:1]: I will make the changes; it's good to stay consistent. I'll check everyone's files and make them lowercase if they are already not.

**Pull request merged**